### PR TITLE
feat: time agnostic APIs

### PR DESCRIPTION
**Pull request by jake-bassett**
## Description
Time agnostic APIs

### CODE REVIEW

File: projects/distributed-tracing/src/shared/components/table/data-cell/metric/metric-table-cell-parser.ts [hunk 1]
```diff
@@ -5,20 +5,23 @@ import { TracingTableCellType } from '../../tracing-table-cell-type';
 @TableCellParser({
   type: TracingTableCellType.Metric
 })
-export class MetricTableCellParser extends TableCellParserBase<number, number, number> {
-  public parseValue(cellData: CellData): number {
-    return Math.round(this.extractValue(cellData)!);
+export class MetricTableCellParser extends TableCellParserBase<number, number | null, number | null> {
+  public parseValue(cellData: CellData): number | null {
```
Comment (aaron-steinfeld) [inline projects/distributed-tracing/src/shared/components/table/data-cell/metric/metric-table-cell-parser.ts:9]: Does it make more sense to convert any nulls to undefined? We don't generally want nulls moving through the code, so cutting them off early is good

File: projects/distributed-tracing/src/shared/dashboard/data/graphql/graphql-options-data-source.model.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { TableControlOption } from '@hypertrace/components';
+import { Observable } from 'rxjs';
+import { GraphQlDataSourceModel } from './graphql-data-source.model';
+
+export abstract class GraphqlOptionsDataSourceModel<
```
Comment (aaron-steinfeld) [inline projects/distributed-tracing/src/shared/dashboard/data/graphql/graphql-options-data-source.model.ts:5]: Is this old? I think this got renamed in a previous PR

File: projects/distributed-tracing/src/shared/components/table/data-cell/metric/metric-table-cell-parser.ts (hunk 1 above)
Comment (jake-bassett) [reply projects/distributed-tracing/src/shared/components/table/data-cell/metric/metric-table-cell-parser.ts:9]: Absolutely. Much better.

File: projects/distributed-tracing/src/shared/dashboard/data/graphql/graphql-options-data-source.model.ts (hunk 2 above)
Comment (jake-bassett) [reply projects/distributed-tracing/src/shared/dashboard/data/graphql/graphql-options-data-source.model.ts:5]: Good catch.

File: projects/distributed-tracing/src/shared/components/table/data-cell/metric/metric-table-cell-parser.ts [hunk 3]
```diff
@@ -21,12 +21,12 @@ export class MetricTableCellParser extends TableCellParserBase<number, number |
   }
 
   // tslint:disable-next-line:no-null-undefined-union
```
Comment (aaron-steinfeld) [inline projects/distributed-tracing/src/shared/components/table/data-cell/metric/metric-table-cell-parser.ts:23]: this lint rule no longer needed

**Review by aaron-steinfeld — APPROVED** [review]

**Pull request merged**